src/Storages/StorageURL.cpp

```diff
@@ -602,8 +602,7 @@ Pipe IStorageURLBase::read(
     Block block_for_format;
     if (isColumnOriented())
     {
-        columns_description = ColumnsDescription{
-            storage_snapshot->getSampleBlockForColumns(column_names).getNamesAndTypesList()};
+        columns_description = storage_snapshot->getDescriptionForColumns(column_names);
```

Comment:
What about `StorageURLWithFailover::read(...)`?

Reply:
Didn't noticed it, fixed. Thanks!